lib/playlist-generation/synthesize-playlist.ts

```diff
@@ -0,0 +1,69 @@
+import { AudioFeatures, Song, User } from '..';
+import { getNRandomElements, kLargest } from '../../utils';
+import { Lobby } from '../lobby';
+import { addSongs } from '../spotify/add-songs';
+import { THEME, theme2Conditions } from './themes';
+
+/**
+ *
+ * @param lobbyId the lobby to analyze
+ */
+export async function synthesizePlaylist(lobbyId: string, theme: THEME): Promise<boolean> {
+  const lobby = await Lobby.fromId(lobbyId);
+  if (!lobby) return false;
+
+  const songsMap = await lobby.participants.reduce(async (accP: Promise<Record<string, number>>, userId) => {
+    const acc = await accP;
+
+    const user = await User.fromId(userId);
+    if (!user) return acc;
+
+    user.topSongs.forEach((songId) => {
+      if (!acc[songId]) acc[songId] = 0;
+      acc[songId]++;
+    });
+
+    return acc;
+  }, Promise.resolve({}));
```

Comment:
this gets the frequencies of songs among all users in the lobby?

Reply:
yup! it's a frequency map